PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/IDEHelperImpl.java

```diff
@@ -393,6 +409,69 @@ public void openAppServiceFile(final AppServiceFile file, Object context) {
         AzureTaskManager.getInstance().runInModal(task);
     }
 
+    private boolean openFileInEditor(final AppServiceFile appServiceFile, VirtualFile virtualFile, FileEditorManager fileEditorManager) {
+        final FileEditor[] editors = fileEditorManager.openFile(virtualFile, true, true);
+        if (editors.length == 0) {
+            return false;
+        }
+        for (FileEditor fileEditor : editors) {
+            if (fileEditor instanceof TextEditor) {
+                final String originContent = getTextEditorContent((TextEditor) fileEditor);
+                final MessageBusConnection messageBusConnection = fileEditorManager.getProject().getMessageBus().connect(fileEditor);
+                messageBusConnection.subscribe(FileEditorManagerListener.Before.FILE_EDITOR_MANAGER, new FileEditorManagerListener.Before() {
```

Comment:
shall we subscribe a listener in a loop?

Reply:
Using loop as `fileEditorManager.openFile` returns `FileEditor[]`, I will check whether there is only one editor for text file and do correspond refactor in later pr. Besides, all listener is bind to correspond `FileEditor` and will be unsubscribe when editor closed.